chalk-solve/src/clauses/builtin_traits/fn_family.rs

```diff
@@ -25,107 +25,96 @@ pub fn add_fn_trait_program_clauses<I: Interner>(
 ) -> Result<(), Floundered> {
     let interner = db.interner();
     let trait_id = db.well_known_trait_id(well_known).unwrap();
+
+    let push_clauses = |builder: &mut ClauseBuilder<'_, I>,
```

Comment:
This is a total style nit and more of a suggestion, but I find really big and complex closures hard to read, and would rather see this pulled out as a private helper function.

Reply:
(I realize it does use a few variables from the environment, so it would require a few extra parameters; I think I just prefer to see those parameters so I can tell just what it needs, and also digest each function more in isolation. If it's a *lot* of parameters then I would try to use methods somehow.)